### PR TITLE
checks if unit is alive after status effects.

### PR DESCRIPTION
**Pull request by IceDune**
modified the performTurn so that effects are taken into consideration.

### CODE REVIEW

File: packages/warriorjs-core/src/Unit.js [hunk 1]
```diff
@@ -92,10 +92,16 @@ class Unit {
    */
   performTurn() {
     if (this.isAlive()) {
```
Comment (olistic) [inline packages/warriorjs-core/src/Unit.js]: This check is now redundant with the inclusion of the inner `.isAlive()` checks.

File: packages/warriorjs-core/src/Unit.js [hunk 2]
```diff
@@ -92,10 +92,16 @@ class Unit {
...
-      this.effects.forEach(effect => effect.passTurn());
-      if (this.turn.action && !this.isBound()) {
-        const [name, args] = this.turn.action;
-        this.abilities.get(name).perform(...args);
+      this.effects.forEach(effect => {
```
Comment (olistic) [inline packages/warriorjs-core/src/Unit.js]: This loop can be optimized and exit the `performTurn` method early if the `.isAlive()` check fails for one event.

**Comment by IceDune** [comment]
hi @olistic , may i know what was wrong in the way i PR'd. is it about the tests or is it about how i did the PR?
thank you.

File: packages/warriorjs-core/src/Unit.js (hunk 2 above)
Comment (olistic) [reply packages/warriorjs-core/src/Unit.js]: You could use `Array.some()`, but that'd lead to more obscure code. You can use `for..of` instead and just exit early with a `return` statement (`this.effects` is an instance of [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map#Iterating_Maps_with_for..of)).

**Comment by olistic** [comment]
Hello @IceDune. It was not about "how" you opened the PR, it's about the tests (more specifically linting and formatting issues). Note that CI is failing, if you click on details you will see why. But more importantly, make sure you run the tests locally before submitting code for review. Please read and make sure you are following the [CONTRIBUTING guidelines](https://github.com/olistic/warriorjs/blob/master/CONTRIBUTING.md). Thanks.

File: packages/warriorjs-core/src/Unit.js [hunk 3]
```diff
@@ -92,12 +92,13 @@ class Unit {
    */
   performTurn() {
     if (this.isAlive()) {
-      this.effects.forEach(effect => { 
-        if(this.isAlive()) {
-          effect.passTurn(); 
+      for (let effect of this.effects) {
+        effect.passTurn();
+        if (this.isAlive()) {
```
Comment (IceDune) [inline packages/warriorjs-core/src/Unit.js]: hello @olistic , i have a doubt about this. Is the isAlive check needed after passTurn()? and
you suggested to use return to stop looping. how do i do that? with return -1? I'm a bit confused.
thank you.

**Comment by IceDune** [comment]
hi @olistic , when i run yarn test, it says cannot find files in the path packages/**/src. could you tell me how to solve this? I searched for it but couldnt find any help. Is it due to the way windows handles path unlike linus and unix?

**Comment by IceDune** [comment]
@olistic , please let me know if there is any change that needs to be done in the PR.
thanks a lot

**Comment by olistic** [comment]
Hello @IceDune, sorry for the big delay. I'll be a bit busy for a couple more days, and then I'll review this.

**Pull request closed without merging**